packages/flutter_tools/test/integration.shard/observatory_port_test.dart

```diff
@@ -0,0 +1,116 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/convert.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+import 'test_data/basic_project.dart';
+import 'test_utils.dart';
+
+Future<int> getFreePort() async {
+  int port = 0;
+  final ServerSocket serverSocket = await ServerSocket.bind(InternetAddress.loopbackIPv4, 0);
+  port = serverSocket.port;
+  await serverSocket.close();
+  return port;
+}
+
+Future<void> waitForObservatoryMessage(Process process, int port) async {
+  final Completer<void> completer = Completer<void>();
+  process.stdout
+    .transform(utf8.decoder)
+    .listen((String line) {
+      print(line);
+      if (line.contains('An Observatory debugger and profiler on Flutter test device is available at')) {
+        if (line.contains('http://127.0.0.1:$port')) {
+          completer.complete();
+        } else {
+          completer.completeError(Exception('Did not forward to provided port $port, instead found $line'));
+        }
+      }
+    });
+  process.stderr
+    .transform(utf8.decoder)
+    .listen(print);
+  return completer.future;
+}
+
+void main() {
+  Directory tempDir;
+  final BasicProject _project = BasicProject();
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('run_test.');
+    await _project.setUpIn(tempDir);
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  testUsingContext('flutter run --observatory-port', () async {
```

Comment:
Do these work if you switch them to `testWithoutContext`?

Reply:
Nope :-( Was getting `Unsupported operation: context.get<FileSystem> is not supported in test methods. Use Testbed or testUsingContext if accessing Zone injected values.` so I changed to using `testUsingContext` which seemed to work.